pkg/cloudprovider/aws/fleet/packing/instancetype.go

```diff
@@ -0,0 +1,159 @@
+		// 	},
+		// 	pricePerHour: 0.768,
+		// },
+		{
+			name: "m5.2xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			pricePerHour: 0.384,
+		},
+		{
+			name: "m5.xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			pricePerHour: 0.192,
+		},
+		{
+			name: "m5.large",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			availableCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			pricePerHour: 0.096,
+		},
+	}
+)
+
+type instanceType struct {
+	name string
+	sync.RWMutex
+	availableCapacity v1.ResourceList
+	totalCapacity     v1.ResourceList
+	pricePerHour      float64
+}
+
+func (p *PodPacker) getInstanceTypes(filter string) []*instanceType {
+	instanceTypeOptions := instanceTypesAvailable
+	// sort by increasing capacity of the instance
+	sort.Sort(byCapacity{instanceTypeOptions})
+	return instanceTypeOptions
+}
+
+func (r *instanceType) isAllocatable(cpu, memory resource.Quantity) bool {
+	r.Lock()
+	defer r.Unlock()
+	return (r.availableCapacity.Cpu().Cmp(cpu) >= 0) &&
+		(r.availableCapacity.Memory().Cmp(memory) >= 0)
+}
+
+func (r *instanceType) reserveCapacity(cpu, memory resource.Quantity) error {
+	r.Lock()
+	defer r.Unlock()
+	if r.availableCapacity.Cpu().IsZero() {
+		return fmt.Errorf("insufficient CPU capacity")
+	}
+	if r.availableCapacity.Memory().IsZero() {
+		return fmt.Errorf("insufficient Memory capacity")
+	}
+	r.availableCapacity.Cpu().Sub(cpu)
+	r.availableCapacity.Memory().Sub(memory)
+	return nil
+}
+
+func filterInstancesBasedOnCost(instanceTypeOptions []*instanceType) []*instanceType {
+	sort.Sort(byCost{instanceTypeOptions})
+	return instanceTypeOptions
+}
+
+type instanceTypes []*instanceType
+
+func (c instanceTypes) Len() int {
+	return len(c)
+}
+
+func (c instanceTypes) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+
+type byCapacity struct{ instanceTypes }
+
+func (c byCapacity) Less(i, j int) bool {
+	if c.instanceTypes[i].totalCapacity.Cpu().MilliValue() ==
+		c.instanceTypes[j].totalCapacity.Cpu().MilliValue() {
+		return c.instanceTypes[i].totalCapacity.Memory().MilliValue() <
+			c.instanceTypes[j].totalCapacity.Memory().MilliValue()
+	}
+	return c.instanceTypes[i].totalCapacity.Cpu().MilliValue() <
+		c.instanceTypes[j].totalCapacity.Cpu().MilliValue()
+}
+
+type byCost struct{ instanceTypes }
```

Comment:
I get that this kind of saves a few characters but I think it's not really the style in the project - I'd just leave things as slices.

Reply:
I think this is how you implement comparators in go